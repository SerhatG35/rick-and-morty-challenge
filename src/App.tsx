import { lazy, Suspense } from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import Navbar from './components/Navbar'
import ReturnToTopButton from './components/ReturnToTopButton'
import { ROUTES } from './constants/Routes'
import { GlobalStyles } from './styles/global'
import { MainContainer } from './styles/styles'

const Episodes = lazy(() => import('./pages/Episodes'))
const Locations = lazy(() => import('./pages/Locations'))

function App() {
    return (
        <Router>
            <Suspense fallback={<div>Loading...</div>}>
                <MainContainer data-testid='main-container'>
                    <GlobalStyles />
                    <Navbar />
                    <Switch>
                        <Route exact path={ROUTES.EPISODES} component={Episodes} />
                        <Route path={ROUTES.LOCATIONS} component={Locations} />
                    </Switch>
                    <ReturnToTopButton />
                </MainContainer>
            </Suspense>
        </Router>
    )
}

export default App
