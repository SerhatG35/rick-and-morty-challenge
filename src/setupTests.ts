import '@testing-library/jest-dom/extend-expect'
jest.mock('react-intersection-observer')
jest.mock('src/service/dataFetch')
