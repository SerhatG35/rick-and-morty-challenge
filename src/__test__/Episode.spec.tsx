import '@testing-library/jest-dom/extend-expect'
import { cleanup, render, waitFor } from '@testing-library/react'
import App from 'src/App'
import { mockData } from 'src/fixtures'

afterEach(cleanup)

// jest.mock('axios');
jest.mock('../service/axios')

test('renders components without an error', async () => {
    const wrapper = render(<App />)
    await waitFor(() => wrapper)

    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
})

test('episode loaded correctly', async () => {
    const wrapper = render(<App />)
    await waitFor(() => wrapper)

    expect(wrapper.getByTestId('episode-wrapper')).toHaveTextContent(mockData.name)

    wrapper.unmount()
})
