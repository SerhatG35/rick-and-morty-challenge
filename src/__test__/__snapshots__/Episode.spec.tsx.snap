// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components without an error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM sc-gtsrHT bGXArO eUWZAL"
        style="display: none;"
      >
        <nav
          class="sc-dlnjwi fgzHqQ"
        >
          <a
            class="sc-hKFxyN kuTqTm"
            href="/"
          >
            By Episodes
          </a>
          <a
            class="sc-hKFxyN kWmvFd"
            href="/locations"
          >
            By Locations
          </a>
        </nav>
        <button
          class="sc-fujyAs gMcagR"
          display="none"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M868 545.5L536.1 163a31.96 31.96 0 0 0-48.3 0L156 545.5a7.97 7.97 0 0 0 6 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
            />
          </svg>
        </button>
      </div>
      <div>
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM sc-gtsrHT bGXArO eUWZAL"
      style="display: none;"
    >
      <nav
        class="sc-dlnjwi fgzHqQ"
      >
        <a
          class="sc-hKFxyN kuTqTm"
          href="/"
        >
          By Episodes
        </a>
        <a
          class="sc-hKFxyN kWmvFd"
          href="/locations"
        >
          By Locations
        </a>
      </nav>
      <button
        class="sc-fujyAs gMcagR"
        display="none"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M868 545.5L536.1 163a31.96 31.96 0 0 0-48.3 0L156 545.5a7.97 7.97 0 0 0 6 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
          />
        </svg>
      </button>
    </div>
    <div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
